clustered/client/src/main/java/org/ehcache/clustered/client/internal/ConnectionSource.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.client.internal;
+
+import org.terracotta.connection.ConnectionException;
+import org.terracotta.lease.connection.LeasedConnection;
+import org.terracotta.lease.connection.LeasedConnectionFactory;
+
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+
+public abstract class ConnectionSource {
+
+  public abstract String getCacheManagerName();
+
+  public abstract LeasedConnection connect(Properties connectionProperties) throws ConnectionException;
+
+  public abstract URI getClusterUri();
+
+  public abstract Iterable<InetSocketAddress> getServers();
```

Comment:
I don't think you need to expose this (and certainly not on both implementations).

Reply:
`ConnectionSource` represents the source of the connection, which could either be a URI, or an iterable of servers. I don't understand why having `getServers` in `ConnectionSource` is bad, or unneeded..

Reply:
Because only one of your subclasses can implement it.  If you want you could make it a method on the subclass. It's opposite only exists because it predates the addition.

Reply:
Done